alpha/apps/kaltura/lib/storage/urlTokenizers/kWowzaSecureTokenUrlTokenizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+class kWowzaSecureTokenUrlTokenizer extends kUrlTokenizer
+{
+	
+	/**
+	 * @var string
+	 */
+	protected $paramPrefix;
+	
+	/**
+	 * @var bool
+	 */
+	protected $shouldIncludeClientIp;
+	
+	/**
+	 * @var string
+	 */
+	protected $hashAlgorithm;
+
+	
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		$tokenPrefix = $this->getParamPrefix();
+		$expiryTimeFrame = $this->getWindow();
+		
+		$params = array();
+		
+		$params["starttime"] = time();
+		$params["endtime"] = time() + $expiryTimeFrame;
+		
+		$prefixedParams = $this->prefixParams($tokenPrefix,$params);
+		
+		return $url."?".http_build_query($prefixedParams,'','&').'&'.$tokenPrefix.'hash='.$this->getHash($prefixedParams,$url);
+		
+	}
+	
+	/**
+     * Get prefixed parameters
+     * 
+     * @return array
+     */
+	protected function prefixParams($prefix,$params){
+		
+		if(!empty($prefix))
+        {
+            foreach($params as $key => $param)
+            {
+                if(strpos($key, $prefix) === false)
```

Comment:
Better use a different array for the result, will remove the need for strpos (simply $result[$key] = $prefix . $param)